subsys/net/lib/dns/mdns_responder.c

```diff
@@ -645,55 +646,80 @@ static void setup_ipv4_addr(struct sockaddr_in *local_addr)
 
 static int init_listener(void)
 {
-	int ret, ok = 0;
+	int ret, ok = 0, i;
+	struct net_if *iface;
+	int iface_count;
+
+	NET_IFACE_COUNT(&iface_count);
+	NET_DBG("Setting mDNS listener to %d interface%s", iface_count,
+		iface_count > 1 ? "s" : "");
+
+	if (iface_count > CONFIG_NET_IF_MAX_IPV4_COUNT ||
+	    iface_count > CONFIG_NET_IF_MAX_IPV6_COUNT) {
+		NET_WARN("You have %d interfaces configured but there "
+			 "are %d network interfaces in the system.",
+			 MAX(CONFIG_NET_IF_MAX_IPV4_COUNT,
+			     CONFIG_NET_IF_MAX_IPV6_COUNT), iface_count);
+	}
```

Comment:
Why would we need such a warning here? Isn't it already covered in `iface_ipv6/4_init()` (it'd also print a warning for similiar case)?

Reply:
The warning is similar but context is a bit different. I think it makes sense to print a warning here, otherwise it might be difficult to track this issue if mDNS does not start properly.